src/Application.php

```diff
@@ -639,33 +468,15 @@ private function checkForDuplicateRoute($path, $methods = null)
      */
     private function emitMarshalServerRequestException($exception)
     {
-        $response = (new Response())
-            ->withStatus(StatusCode::STATUS_BAD_REQUEST);
-        $finalHandler = $this->getFinalHandler();
-        $response = $finalHandler(new ServerRequest(), $response, $exception);
+        if ($this->container && $this->container->has(Middleware\ErrorResponseGenerator::class)) {
```

Comment:
This is outside that system - this method is specifically invoked within `run()` only if an exception occurs when attempting to create a `ServerRequest` via the `ServerRequestFactory`; the pipeline is never executed in this situation.